src/lua/zencode_array.lua

```diff
@@ -246,7 +246,7 @@ IfWhen("the '' is found in '' at least '' times", function(ele, arr, times)
 	local found = constructor(0)
 	local one = constructor(1)
 	for _,v in pairs(list) do
-		if v == obj then found = found + one end
+		if type(v) == type(obj) and v == obj then found = found + one end
```

Comment:
this part coul dleverage `:octet()` on `iszen(type( ) )` positive results, if float cast to octet is possible.

Reply:
`:octet()` doesn't see the difference in type. We risk to have the octet which represent the string "1" to be equal to the float 1 (we don't want different type that are equal when compared)